io/net/virt-net/lpm_basic.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+# Co-author: Kalpana Shetty <kalshett@in.ibm.com>
+
+'''
+Tests for Basic Live Partition Mobility
+'''
+
+import time
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils.process import CmdError
+from avocado import skipIf, skipUnless
+
+IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()
+IS_KVM_GUEST = 'qemu' in open('/proc/cpuinfo', 'r').read()
+
+
+class CommandFailed(Exception):
+    '''
+    Defines the exception called when a
+    command fails
+    '''
+
+    def __init__(self, command, output, exitcode):
+        Exception.__init__(self, command, output, exitcode)
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class LPM(Test):
+    '''
+    Performs LPM from source server to remote server, and back.
+    '''
+    @skipUnless("ppc" in distro.detect().arch,
+                "supported only on Power platform")
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST,
+            "This test is not supported on KVM guest or PowerNV platform")
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        self.install_packages()
+        self.rsct_service_start()
+
+        self.hmc_ip = self.get_mcp_component("HMCIPAddr")
+        if not self.hmc_ip:
+            self.cancel("HMC IP not got from lsrsrc command")
+        self.hmc_user = self.params.get("hmc_username", default='hscroot')
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default='abc123')
+
+        self.lpar = self.get_mcp_component("NodeNameList").split('.')[0]
```

Comment:
I don't think so, as long as the lpar hostname is set, this will not fail in SLES

Reply:
> This will Fail for Sles OSes. So please handled Accordingly.

Reply:
> I don't think so, as long as the lpar hostname is set, this will not fail in SLES

for that, we have to add this in Readme file saying change the host name to some formate, because the by default hostname will not work here for sles. and we have to manually change it before running this test.
Or
Narasimhan said he has different way to handle it.
@narasimhan-v  please comment here.

Reply:
I need to check on sles running this same code.